BUILDING.md

```diff
@@ -0,0 +1,68 @@
+# Building JanusGraph client for Python
+
+- Building Library:
+
+A set of scripts have been created to make the job of building the library easier for user, but the scripts
+expects few pre requisites to be installed before hand. 
+
+- Pre-Requisites:
+    - Python needs to be installed on system. The library is currently tested against Python
+    3.4, 3.5 & 3.6, though is expected to work against any Python 3.
+    
+    - Python is added to system path. Make sure, the user can run the following commands from command line:
+        - python3/ python3.4/ python3.5/ python3.6 on Unix machines.
+        - python on Windows machines.
+    
+    - The automated build scripts are tested on following Operating Systems:
+        - Windows 7
+        - CentOS
+        - RHEL 7
+        - Ubuntu 18.04
+        - MacOS
+        
+- Building on Windows 7:
+    - A script build.cmd is created which accepts following keyword arguments:
+        - d: Specifies weather documents to be build. Set to true or false.
+        - b: Specifies weather library to be build. Set to true or false.
+        - i: Specifies weather library to be installed or not. Set to true or false.
+    
+    - If any of parameters is missing, the script exits with error message. 
+    An appropriate way to invoke the script would be:
+        ```bash
+        build.cmd "d=true" "b=true" "i=true"
+        ```
+    - The above example builds the docs, library and also installs to local global Python site-packages.
+
+- Building on Unix System:
+    - A script named build.sh is created which accepts following parameters:
+        - -d: Default to true. Implies, if documentation is to be build.
+        - -b: Defaults to true. Implies if library is to be build.
+        - -i: Default to false. Implies if library is to be installed on local system.
+        - -p: Defaults to 3.6. This specifies the Python 3 version installed on system. You can check 
+        the parameter to pass here quickly, by running either of following "python3 --version" or "python3.4 --version"
+        or "python3.5 --version" etc and passing accoringly.
+        
+    **NOTE**: If you are building library on Max OS (OSx), then you will need to run by pre pending sudo -H. Eg: sudo -H ./build.sh
+    
+    If any of the defaults needs to be changed, the parameter needs to be specified explicitly while running the script.
+    
+     - If you want to build just the documentation, without the library.
+        ```bash
+        ./build.sh -b false
```

Comment:
This is not the typical way that Unix/Linux command line flags are written. It's typically `-b` implies true, and `--no-b` or `--without-b` implies false.

Can we change this to:

* `-d`, `--docs`: build docs
* `--no-docs`: does not build docs
* `-l`, `--library`: build library
* `--no-library`: does not build library
etc.

Thoughts?

Reply:
Please see my my previous comment.